pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -218,6 +220,23 @@ func (m *MigCluster) GetClusterSubdomain(c k8sclient.Client) (string, error) {
 	return clusterSubdomain, nil
 }
 
+// GetRegistryImage gets a MigCluster specific registry image from ConfigMap
+func (m *MigCluster) GetOperatorVersion(c k8sclient.Client) (string, error) {
+	clusterClient, err := m.GetClient(c)
+	if err != nil {
+		return "", liberr.Wrap(err)
+	}
+	clusterConfig, err := m.GetClusterConfigMap(clusterClient)
+	if err != nil {
+		return "", liberr.Wrap(err)
+	}
+	operatorVersion, ok := clusterConfig.Data[OperatorVersionKey]
+	if !ok {
+		return "", liberr.Wrap(errors.Errorf("configmap key not found: %v on MigCluster %v. Ensure that all MigClusters are running the same version of MTC Operator.", OperatorVersionKey, m.ClusterName))
```

Comment:
I actually wonder if we should raise a critical condition `ClusterConfigmapKeyNotFound` when this occurs so that it's visible in the UI when this key can't be found, since it's going to be the case that everyone who gets autoupgraded to 1.4.2 or whatever will have an older MTC version on their 3.x cluster and will hit this error and need to dig through logs to fix.